roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,269 @@
+
+### Bidding for a slot/Auctioning System
+
+#### Goals
+
+1. Price finding
+2. Be efficient ... we are talking about slots for single blocks, the overhead
+   of assignment should be minimal.
+3. Maximize utilization: Avoid empty slots, as long as there is demand from
+   parathread willing to pay the minimum price.
+4. DoS resistance: We should not be able to get flooded with dubious bids, that
+   just fill up runtime resources.
+5. Lifetime/Update: Bids should have some lifetime or alternatively need to be
+   upgradeable. E.g. someone puts in a bid, but it is too low and is constantly
+   outbid by other bids ... eventually we either no longer care ... would like
+   to remove the bid or increase it, so it will finally make it.
+6. The user needs to know whether their bid made it through or not and when.
+
+#### General Interface
+
+The user can apply for a slot by placing a bid extrinsic. This bid will contain
+some price the user is willing to pay for a core. These extrinsic go into a
+transaction pool, as other extrinsics.
+
+There should be no reason to place bids long before we want to claim a slot. If
+spam protection mechanism are effective you should just send the bid, when you
+want to have a core in the near future ... bids will have some lifetime
+specified. If they don't manage to win a slot within that lifetime, they are
+dropped. The user would then need to bid again.
+
+Needs:
+
+1. Expose: Realistic price to pay right now. (E.g. expose recent max/lowest
+   prices/average).
+2. Actual utilization - related to price, but nevertheless good to know
+   explicitly. If there are lots of free cores right now, it is "safe" to go with
+   the minimum bid.
+3. Some form of lifetime ... Some time frame the user can expect to either get a
+   core or know for sure he won't get any. Needs to be exposed somehow? - HOW?
+4. The user does not need to be able to specify a start ... once he submits a
+   bid, we treat this as "get me a core asap". No need to be able to schedule in
+   advance.
+
+I think it is fine for the lifetime to be implicit (determined by the system,
+e.g. by a 10 block auction mechanism), but we need to expose what happened to
+the bid. The actual auction mechanism used, should not be exposed.
+
+#### Node Side Implementation
+
+On the node side validators should be able to do some pre-processing on bids,
+for handling spam. In particular this is only relevant, if there are lots of
+bids. Ideally on the node side we would already:
+
+1. Check signatures, remove invalid bids and disconnect peers providing such
+   bids.
+2. If a certain threshold of pending bids is reached, we can start dropping
+   lowest bids.
+
+TODO: How does this work with transactions? Can we even check signatures on the
+node side? If we don't check signatures on the node side, then how can we be
+spam resistant?
+
+Then the block producer should put bids into the block, limited by some maximum
```

Comment:
This implies that the bidding is basically off-runtime. What's the justification to handle this? I presume the parachain protocol has made similar decisions in other places as well.

Reply:
Basically weight limit. Block producer can not just put everything that is available in there, no matter what. But we already revisited how bidding works, so this is pretty much obsolete already.